### DISCUSSION TITLE
DataflowBroadcast around DataflowStream[?] error

### DISCUSSION THREAD

**Discussion by FelipeMSD** [nextflow-io/nextflow#4576] [Q&A] [answered]
Dear all, I don't know why this is happening now, because I didn't have this problem in the past, but when I use `variable=Channel.of("string")`  and then `variable.view()`  Nextflow returns me `**DataflowBroadcast around DataflowStream[?]**` . Has this ever happened to you? I'm even using `nextflow.enable.dsl=2` in the begining of my main.nf file but it seems to not help...

**Comment by bentsherman** [comment]
Can you share the exact script you are running and your Nextflow version?

**Reply by FelipeMSD** [reply]
Hi @bentsherman . thanks for the prompt reply. For now I cannot share it but I made few tests and it seems the problem is with `.map{[]}`, as now I also had this message: `WARN: Cannot serialize context map. Cause: java.lang.IllegalArgumentException: Unable to create serializer "com.esotericsoftware.kryo.serializers.FieldSerializer" for class: jdk.internal.reflect.DelegatingMethodAccessorImpl -- Resume will not work on this process`. When I run `variable.view()` I can see the values but when I try `x=variable1.map{[variable2]}` and then `x.view()` what I have is: [variable1_string, DataflowBroadcast around DataflowStream[?]]. Any insight on this? ChatGPT told me to update Nextflow, which I did; check if my java version is higher than 8, which it is, and also try disabling the Kryo serialization in Nextflow by adding `process.executor = 'local'` to my script. Unfortunately, it doesn't seem to help me... It only happens for the `Channel.of()` channels. The `file()` channels appear on my `.map{[]}` list correctly.

**Reply by FelipeMSD** [reply]
I've been doing some tests here and I decided to call the parameter inside the .map{[]} array as `x=variable1.map{[params.variable2]}` instead. It's working this way now. :wink:

**Reply by bentsherman** [reply, marked as answer]
Is `variable2` another channel in your workflow? If so that is likely the problem, you cannot use channels inside the closure of a channel operator like that

**Reply by FelipeMSD** [reply]
I see... So I'll keep calling variable2 this way: `x=variable1.map{[params.variable2]}`. Thank you very much for the insights ;)
